probe-rs/src/architecture/arm/core/armv7m.rs

```diff
@@ -1019,6 +1020,25 @@ impl<'probe> CoreInterface for Armv7m<'probe> {
     fn fpu_support(&mut self) -> Result<bool, crate::error::Error> {
         Ok(self.state.fp_present)
     }
+
+    fn enable_vector_catch(&mut self, vector_catch: VectorCatchCondition) -> Result<(), Error> {
```

Comment:
once enabled, there's no API to disable it. Perhaps we can do instead `set_vector_catch` with a struct with `hard_fault: bool, core_reset: bool` so you can set it to (false, false) to disable?

Reply:
The bool-api is a bit unflexible imo, because armv7m can technically have 6 (I think) different exceptions whereas armv6 only has the two. Maybe we just need a `disable_vector_catch`? Tho what do we do with `DBGEN`?

Reply:
so if we add support for all the exceptions  in v7 in the future, the enum will have 63 variants? :)

Reply:
ah no, I misunderstood. I see that it's only enabling the requested bits, so you can call it multiple times to enable the separate vectors, right? The `All` variant confused me, I thought you'd need all `2^n` combinations in the variant.

Having a `disable_vector_catch` sounds good to me. No idea what to do about `debugen`. perhaps leave it as-is? or make it 0 if all the vector catch bits are back to zero?

Reply:
I decided to not disable it at all because it has weird sideeffects that someone might not expect when they turned on debug mode and it's not disabled for them.

Enabling is no problem because it cannot work without it so it is expected.